openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -338,3 +348,58 @@ def get(self, request, course_key_string, *args, **kwargs):
             pass
 
         return Response(resp)
+
+
+class Celebration(DeveloperErrorViewMixin, APIView):
+    """
+    **Use Cases**
+
+        Marks a particular celebration as complete
+
+    **Example Requests**
+
+        POST /api/courseware/celebration/{course_key}
+
+    **Request Parameters**
+
+        Body consists of the following fields:
+
+            * first_section (bool): when a user finishes their first section of a course
+
+    **Returns**
+
+        * 200 on success with above fields.
+        * 400 if an invalid parameter was sent.
+        * 404 if the course is not available or cannot be seen.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated, )
+    http_method_names = ['post']
+
+    def post(self, request, course_key_string, *args, **kwargs):
+        """
+        Handle a POST request.
+        """
+        course_key = CourseKey.from_string(course_key_string)
+
+        data = dict(request.data)
+        first_section = data.pop('first_section', None)
```

Comment:
Nit: move this down to right before you use the `first_section` variable.

Reply:
We pop it here, so that the next line (`if data:`) can see whether there's any passed in parameters that we don't know about.

Reply:
Great call. I missed that.